pkg/patterns/addon/pkg/loaders/fs.go

```diff
@@ -66,29 +65,12 @@ func (c *ManifestLoader) ResolveManifest(ctx context.Context, object runtime.Obj
 		componentName string
 	)
 
-	unstruct, ok := object.(*unstructured.Unstructured)
-	if ok {
-		v, _, err := unstructured.NestedString(unstruct.Object, "spec", "version")
-		if err != nil {
-			return nil, fmt.Errorf("unable to get spec.version: %v", err)
-		}
-		version = v
-
-		c, _, err := unstructured.NestedString(unstruct.Object, "spec", "channel")
-		if err != nil {
-			return nil, fmt.Errorf("unable to get spec.version: %v", err)
-		}
-		channelName = c
-
-		componentName = strings.ToLower(unstruct.GetKind())
-	} else if addonObject, ok := object.(addonsv1alpha1.CommonObject); ok {
-		componentName = addonObject.ComponentName()
-
-		spec := addonObject.CommonSpec()
-		version = spec.Version
-		channelName = spec.Channel
-	} else {
-		return nil, fmt.Errorf("object %T was not an addonsv1alpha1.CommonObject", object)
+	version, err := utils.GetCommonVersion(object)
+	componentName, err = utils.GetCommonName(object)
+	channelName, err = utils.GetCommonChannel(object)
+	// All the errors above are checking the same thing? Is there a better way?
```

Comment:
There is a pattern that is sometimes used in parsing https://blog.golang.org/errors-are-values

However in this case because the methods are different it's better just to repeat the `if err != nil` block 3 times.

It might be we combine GetCommonName / GetCommonVersion / GetCommonChannel into GetCommonSpec in future, in which case we'd only have one error check.

Reply:
Great article! I always wondered about the frequency of `if err != nil `.